src/app/frontend/replicasetlist/replicasetlist_state.js

```diff
@@ -32,15 +33,32 @@ export default function stateConfig($stateProvider) {
       replicaSets: resolveReplicaSets,
     },
     templateUrl: 'replicasetlist/replicasetlist.html',
+    onEnter: initState,
   });
 }
 
 /**
+ * Avoids entering replica set list page when there are no replica sets.
+ * Used f.e. when last replica set gets deleted.
+ * Transition to: zerostate
+ * @param {!ui.router.$state} $state
+ * @param {!angular.$timeout} $timeout
+ * @param {!backendApi.ReplicaSetList} replicaSets
+ * @ngInject
+ */
+function initState($state, $timeout, replicaSets) {
+  if (replicaSets.replicaSets.length === 0) {
+    // allow original state change to finish before redirecting to new state to avoid error
+    $timeout(() => { $state.go(zerostate); });
```

Comment:
I've been looking for a better solution but haven't found any. Related issue:

https://github.com/angular-ui/ui-router/issues/1434
https://github.com/angular-ui/ui-router/issues/1234

Here i've found this solution:
http://stackoverflow.com/questions/28603620/cannot-read-property-of-null-when-using-state-in-onenter

Unless we move it to controller as it was, I've no other clean solution for this.